src/lib/core/exception.cc

```diff
@@ -59,6 +59,13 @@ exception_log(struct error *error)
 	e->log();
 }
 
+static struct error *
+exception_dup(const struct error *error)
+{
+	Exception *e = (Exception *) error;
+	return e->dup();
```

Comment:
Since this function can be called from C code, it must not raise an exception. Recently, we [decided](https://github.com/tarantool/tarantool/issues/3534) to panic on heap allocation failures so you can use `xmalloc` for error allocations. Alternatively, you can catch a memory allocation exception here or return `out_of_memory` error if `malloc` fails, like `BuildAlloc` does.

Reply:
Done. Now `exception_dup` returns a copy of the error or the error that caused fail.